src/components/modals/RateDialogsModal.tsx

```diff
@@ -237,9 +315,49 @@ class RateDialogsModal extends React.Component<Props, ComponentState> {
         }
     }
 
+    // Keep track of render height for each activity.  This allows us to
+    // align acitivities across multiple webchat windows by adding padding
+    @autobind
+    onActivityHeight(sourceName: string, index: number, height: number): void {
+    
+        // Find height for this item
+        let activityHeight = this.state.activityHeights.find(ac =>
+            ac.sourceName === sourceName && ac.index === index)
+
+        if (activityHeight) {
+            // If height hasn't been set
+            if (!activityHeight.height) {
+                // Set state via function so events don't clobber each other
+                this.setState(prevState => {
+                     // Update height
+                    activityHeight = prevState.activityHeights.find(ac =>
+                        ac.sourceName === sourceName && ac.index === index)!
+                    activityHeight.height = height
+                    return {activityHeights: prevState.activityHeights}
+                })
+            } 
+        }
+    }
+
+    getRenderData(): RenderData[] {
+
```

Comment:
nit: formatting

Reply:
what formatting?

Reply:
you have extra blank lines, looks out of place. If we want to make a rule that all methods have blank lines under them we should do it to all of them, other wise we don't do it to any of them. People need to be able to recognize patterns otherwise they just do whatever

Reply:
Ah, I see.   Got confused as it's already fixed on my local copy